### PR TITLE
Add a hook to modify the number of cards gained.

### PR DESCRIPTION
**Pull request by pyrrhicPachyderm**
To aid in implementing Mentor.

### CODE REVIEW

File: CUSTOM.md [hunk 1]
```diff
@@ -393,3 +393,11 @@ This section is meant for users who are familiar with both the tts mod and progr
       - `isGain`: **bool** - true if the operation is a gain, false if it's a pay
       - `isUndo`: **bool** - true if the operation is being undone (right click), false otherwise
       - `amount`: **number** - amount of energy gained/paid (before bargain deductions); positive for gain or pay
+- Trigger to modify the number of cards gained
+  - Create object and tag with "Modify Card Gain"
+  - `modifyCardGain(params)`
+    - `params`: **table** - contains data about the player and the cards played
+      - `color`: **string** - is the color of the player whose card cost is being calculated
+      - `minor`: **boolean** - true if gaining minors, false if gaining majors
```
Comment (iakona) [inline CUSTOM.md:401]: for power progression i believe `true` represents major so for consistency sake would probably be best to invert the boolean here

File: CUSTOM.md [hunk 2]
```diff
@@ -393,3 +393,11 @@ This section is meant for users who are familiar with both the tts mod and progr
...
+      - `count`: **table** - the number of cards being gained
```
Comment (iakona) [inline CUSTOM.md:402]: both this and the return are `number` right and not tables?

File: CUSTOM.md (hunk 1 above)
Comment (iakona) [reply CUSTOM.md:401]: even though the boolean before was whether it was minor or not, but now that it's public facing should probably be consistent

File: script.lua [hunk 3]
```diff
@@ -2020,7 +2020,7 @@ function MajorPowerC(obj, player_color, alt_click)
     if alt_click then
         cards = 2
     end
-    startDealPowerCards(false, Player[player_color], cards)
+    startDealPowerCards({minor = false, player = Player[player_color], count = cards})
```
Comment (iakona) [inline script.lua:2023]: matching the ordering of `modifyCardGain` the boolean parameter should be the middle argument

File: CUSTOM.md (hunk 1 above)
Comment (pyrrhicPachyderm) [reply CUSTOM.md]: I've changed it to `true` representing a major throughout the card gain code, for maximum consistency.

**Comment by iakona** [comment]
while only tangentially related to your change, coders now have the ability to gain cards anywhere from 1 to 6 rather than specifically 2, 4 or 6. The code to handle drawing cards from playtest deck vs normal deck will need to be updated to be more robust. i somewhat recently had to do a similar thing with exploratory days that never were, so should more or less be copying the math formulas over into here

for reference here's the fractured code https://github.com/spirit-island/spirit-island-mod/blob/beta/objects/0e627b/contained/fddf31/script.lua#L31

**Comment by pyrrhicPachyderm** [comment]
Copying those formulae exactly would be a behaviour change in the case of drawing two cards with playtest powers set to 3-1. Exploratory Fractured Days uses `floor(count/3)` in this case, which is 0 when gaining two cards, while the current behaviour still draws 1 playtest power in this case. I'm going to say this is a bad change to make, as it means that playtesting using Mentor and 3-1 would never see any playtest cards.

As such, I'm going to put a minimum of one playtest power in. This'll make the new behaviour equivalent to the old for 2, 4, or 6 power cards. That does leave the question of whether the minimum should apply when just drawing one card. Either zero or one playtest powers seems dissatisfactory here ('cause they mean you'll never see the other), so I think I'll randomise it. Let me know if you think that's too complicated.

I'll also update Fractured Days to have the same logic, for consistency.

**Comment by iakona** [comment]
for 1 card, i think it's fine having it always be one. iirc i had a conversation with Ironeye, Ted, or some other developer about this back when i was originally implementing the playtesting options. and the thinking was for something like an element check (aka event) you would pull from the normal deck, but if you need a random effect or what not (i guess in this case trickster) you should pull from playtest

**Comment by pyrrhicPachyderm** [comment]
Thanks; it's good to have dev input on the matter. Change made.

**Pull request merged**